JPF/bad_buildings.sql

```diff
@@ -0,0 +1,284 @@
+# v1.0 - JP Freeley - 11/26/2016
+#
+# SQL - for bad buildings
+# Does not account for DATES or TIMES or AGE. 
+# Includes by building/address counts of 
+#	litigations
+#	hpd_violation 'classes'
+#	hpd_complaints
+#	dob_permits
+#	dob_violations
+# DOB Data joined by ADDRESS .. not sure if I trust BIN nor BBL
+# There are probably more optimal ways to do this.
+# !!! Needs better documentation
+# !!! Ends up with 2000 MORE BUILDINGS THAN EXIST IN hpd_buildings .. likely dirty addresses .. FIRST()??
+### This finds them (1955 rows)
+### select hbaddress, buildingid, count(buildingid) count from bad_buildings group by 1,2 having count > 1;
+
+
+## BELOW REQUIRES THAT NO INDEXES/KEYS PRE-EXIST ON THE TABLES
+## AT THE BOTTOM OF THIS FILE IS CODE FOR REMOVING THE FOLLOWING INDEXES/KEYS
+alter table hpd_buildings add primary key(buildingid);
+alter table hpd_complaintsProb add INDEX(majorcategoryid);
+alter table hpd_complaintsProb add INDEX(minorcategoryid);
+alter table hpd_complaintsProb add INDEX(codeid);
+alter table hpd_complaintsProb add INDEX(complaintid);
+alter table hpd_complaints add INDEX(complaintid);
+alter table hpd_complaints add INDEX(buildingid);
+alter table hpd_violations add INDEX(buildingid);
+alter table hpd_violations add index(class);
+
+
+delete from dob_violations where boro not in ('1','2','3','4','5');
+ALTER TABLE `dob_violations` CHANGE `boro` `boro` INT(5)  NULL  DEFAULT NULL;
+alter table dob_violations add index(boro);
+
+################
+drop table if exists boro_lookup;
+################
+create temporary table boro_lookup select boro, boroid from hpd_buildings group by 1,2;
+alter table boro_lookup add index(boroid);
+
+################
+drop table if exists hs_complaint_count;
+################
+create temporary table 
+	hs_complaint_count
+select	c.buildingid,
+	count(*) as count
+from 
+	hpd_complaintsProb as cp
+left join 
+	hpd_complaints c
+on
+	c.complaintid = cp.complaintid
+left join 
+	hpd_buildings b
+on 
+	c.buildingid = b.buildingid
+where 
+	majorcategoryid  in (12, 59, 65) or
+	minorcategoryid in (92, 196, 112, 375, 67, 323, 381, 106) or 
+	codeid in (800, 801, 802, 803, 804, 809, 1358, 1360, 2713, 2715, 2716, 
+	2833, 2490, 605, 616, 617, 629, 634, 640, 801, 1271, 2400, 2532, 2534, 
+	2661, 2724, 2764, 2767, 2769, 2794) 
+group by 
+	1;	
```

Comment:
Ah, got it. I'm not actually familiar with the `GROUP BY {index}` syntax. (Had to look it up). 

For others, it may be helpful if you clarify the syntax to `group by c.buildingid`